docs/ContentTypeParser.md

```diff
@@ -71,3 +71,41 @@ fastify.addContentTypeParser('*', function (req, done) {
 ```
 
 In this way, all of the requests that do not have a corresponding content type parser will be handled by the specified function.
+
+This is also useful for piping the request stream. You can define a content parser like
+
+```js
+fastify.addContentTypeParser('*', function (req, done) {
+  done()
+})
+```
+
+and then access the core HTTP request directly for piping it where you want:
+
+```
+app.post('/hello', (request, reply) => {
+  reply.send(request.req)
+})
+```
+
+Here is a complete example that pipes images to another destination (e.g. cloud storage):
+
+```js
+fastify.addContentTypeParser('*', (req, done) => {
+    done();
+});
+
+fastify.route({
+    method: 'POST',
+    url: '/api/images/new',
+    handler: (req, res) => {
+        var data = [];
+        req.req.on('data', chunk => data.push(chunk));
+        req.req.on('end', () => {
+            req.body = Buffer.concat(data);
+            // do something with the body such as transforming it to base64
+            // and uploading to your cloud storage
+        });
+    }
+ ```
+});
```

Comment:
This example may be wrong: in fact you can use https://www.fastify.io/docs/latest/ContentTypeParser/#body-parser directly. Maybe we can add and example to parse a [json line](http://jsonlines.org/):

```js
const split2 = require('split2')

  fastify.addContentTypeParser('*', (req, done) => {
    done(null, pump(req, split2(JSON.parse)));
  });

	fastify.route({
    method: 'POST',
    url: '/api/log/jsons',
    handler: (req, res) => {
        var data = [];
        req.body.on('data', d => console.log(d)) // log every incoming object
    }
});

```

or something like that

Reply:
I think we should provide an upload example since I feel like most people will be looking for things like `uploading files to S3` so it can directly provide them the answer but it is up to you.

Maybe we can fix that example using the approach you provided?

Reply:
There is actually [this plugin](https://github.com/fastify/fastify-multipart) for upload so I guess we can give a reference there for upload